### PR TITLE
Rdm 5673 change facets to checkboxes

### PR DESCRIPTION
**Pull request by adammckenzie**
Changes the facets to use checkboxes instead of straight links and change their look and feel. I've also fixed the search details to reflect the search query and facets.

### CODE REVIEW

**Comment by pdante-ubc** [comment]
The facets look good and the checkboxes check, but when I deploy on my system, the facets don't actually seems to affect the search results or the status of the "Clear All" button.

File: app/components/blacklight/facet_item_component.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    # @param [Blacklight::Solr::Response::Facets::FacetItem] item
+    # @param [Hash] options
+    # @option options [Boolean] :suppress_link display the facet, but don't link to it
+    # @return [String]
+    # @private
+    def render_facet_value
+      content_tag(:span, class: "facet-label") do
+        if not @suppress_link
+          link_to(@href, class: "facet-select render-facet-value") do
+            check_box_tag(@label, @label, false, class: "facets-checkbox", :onclick => 'location.href="' + @href + '"') + @label
```
Comment (adammckenzie) [inline app/components/blacklight/facet_item_component.rb:79]: I've added a onclick event to the checkbox as well to make sure that in all browsers clicking on link versus checkbox adds or removes the facet

File: app/components/blacklight/facet_item_component.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ##
+    # Standard display of a SELECTED facet value (e.g. without a link and with a remove button)
+    # @see #render_facet_value
+    # @param [Blacklight::Solr::Response::Facets::FacetField] facet_field
+    # @param [String] item
+    # @private
+    def render_selected_facet_value
+      content_tag(:span, class: "facet-label") do
+        link_to(@href) do
+          check_box_tag(@label, @label, true, class: "facets-checkbox", :onclick => 'location.href="' + @href + '"') + @label
```
Comment (adammckenzie) [inline app/components/blacklight/facet_item_component.rb:96]: Ditto here added the onclick event.

**Comment by pdante-ubc** [comment]
Oh, another issue, although this is a GBL issue, not an Adam code issue. When I select something in a facet, it sometimes moves the map to Asia. Seems to happen when the map view includes the International Dateline (Antimeridian). I checked and this happens on the current FRDR website as well. We should probably see if GBL has fixed this in newer versions of the code.

**Comment by pdante-ubc** [comment]
Oh, that facet not reappearing thing is a GBL issue.

**Comment by adammckenzie** [comment]
> Bug: If I move the map on the results page to an area with no results, the facets all disappear. This is fine since there are no results there, but when I move back to an area with results the facets do not reappear. This happens both on Chrome and Firefox.

Still thinking about this one. Geoblacklight / Blacklight hides the facets when there are no results. This is fine for text search as you can always hit back but it is annoying when a map move hides all of the results as it could be a lot more backs. I tried multiple locations all over the earth on dev 1 which has all of the data from prod from a couple of months ago and I couldn't find a place on earth where we don't show at least one result with the facets disappearing. Any idea how to determine in our production data if there is anywhere in our map where a single geo data result wouldn't be found? If we can't find one this might not be worth overriding the default blacklight / geoblacklight behaviour as the bug wouldn't be experienced by users.

> 
> Functionality: The "Clear All" button tales you back to the landing page. I think it should just clear all the facet selections but leave you on the results page with the map unmoved.

Aha another bug! I wasn't copying all of the query string search parameters properly when removing the facet ones. I think with this new approach it should be an exhaustive list of the other parameters (I am taking the whole and just removing the facet one). Could you give it another try and let me know what you think?

**Comment by adammckenzie** [comment]
The facets now come back if you move the map back or use the clear all button. I have filed: https://computecanada.atlassian.net/browse/RDM-6003 to try to get the facets to stay around when there are no results. Did you file a ticket for the dateline bug you found? I don't think that is in scope for this pull request.

**Pull request merged**